### PR TITLE
Don't save kb when knowledge hasn't been updated

### PR DESCRIPTION
**Pull request by fortunac**
With this update, we saved ~40 seconds on our tar example.

### CODE REVIEW

File: wp/plugin/lib/utils.ml [hunk 1]
```diff
@@ -14,6 +14,7 @@
 open !Core_kernel
 open Bap_main
 open Bap.Std
+open Regular.Std
```
Comment (philzook58) [inline wp/plugin/lib/utils.ml:17]: How did we gain this?

Comment (fortunac) [reply wp/plugin/lib/utils.ml:17]: I opened Regular (http://binaryanalysisplatform.github.io/bap/api/odoc/regular/Regular/Std/index.html) so I can use `Data.Cache.Digest.pp` which I added to the info logs.

**Comment by philzook58** [comment]
This branch:
```
Seq	Host	Starttime	JobRuntime	Send	Receive	Exitval	Signal	Command
1	:	1600887426.345	    25.655	0	0	2	0	./run_single.sh .plt.got > results/.plt.got
6	:	1600887426.353	    26.281	0	0	0	0	./run_single.sh __ctype_tolower_loc > results/__ctype_tolower_loc
3	:	1600887426.349	    26.427	0	0	0	0	./run_single.sh __assert_fail > results/__assert_fail
4	:	1600887426.350	    26.755	0	0	0	0	./run_single.sh __ctype_b_loc > results/__ctype_b_loc
5	:	1600887426.351	    26.766	0	0	0	0	./run_single.sh __ctype_get_mb_cur_max > results/__ctype_get_mb_cur_max
2	:	1600887426.347	    27.140	0	0	0	0	./run_single.sh __argmatch_die > results/__argmatch_die
8	:	1600887452.637	    24.638	0	0	2	0	./run_single.sh __ctype_toupper_loc@plt-0x10 > results/__ctype_toupper_loc@plt-0x10
7	:	1600887452.006	    25.596	0	0	0	0	./run_single.sh __ctype_toupper_loc > results/__ctype_toupper_loc
11	:	1600887453.121	    25.325	0	0	0	0	./run_single.sh __fpending > results/__fpending
10	:	1600887453.109	    25.597	0	0	0	0	./run_single.sh __errno_location > results/__errno_location
```
Master:
```
Seq	Host	Starttime	JobRuntime	Send	Receive	Exitval	Signal	Command
1	:	1600887712.379	    46.378	0	0	2	0	./run_single.sh .plt.got > results/.plt.got
5	:	1600887712.388	    47.644	0	0	0	0	./run_single.sh __ctype_get_mb_cur_max > results/__ctype_get_mb_cur_max
3	:	1600887712.384	    47.873	0	0	0	0	./run_single.sh __assert_fail > results/__assert_fail
6	:	1600887712.391	    48.174	0	0	0	0	./run_single.sh __ctype_tolower_loc > results/__ctype_tolower_loc
4	:	1600887712.386	    48.335	0	0	0	0	./run_single.sh __ctype_b_loc > results/__ctype_b_loc
2	:	1600887712.382	    48.750	0	0	0	0	./run_single.sh __argmatch_die > results/__argmatch_die
8	:	1600887760.034	    45.606	0	0	2	0	./run_single.sh __ctype_toupper_loc@plt-0x10 > results/__ctype_toupper_loc@plt-0x10
7	:	1600887758.767	    47.868	0	0	0	0	./run_single.sh __ctype_toupper_loc > results/__ctype_toupper_loc
10	:	1600887760.569	    46.431	0	0	0	0	./run_single.sh __errno_location > results/__errno_location
11	:	1600887760.725	    46.730	0	0	0	0	./run_single.sh __fpending > results/__fpending
12	:	1600887761.137	    46.574	0	0	0	0	./run_single.sh __fprintf_chk > results/__fprintf_chk
9	:	1600887760.260	    59.394	0	0	0	0	./run_single.sh __do_global_dtors_aux > results/__do_global_dtors_aux
```

I'm not seeing 40s. It does seem like a significant improvement though. I believe those remaining 25s are all load time.

**Comment by philzook58** [comment]
This seems to work and make sense. There is still some performance and RAM usage gains to be had by being more careful about our loading and saving to the cache, but this is a significant improvement.

**Pull request merged**